### ISSUE TITLE
[Feature]: Enhanced Reporting for A11Y violations as well as running playwright in UI Mode

### ISSUE THREAD

**Issue by mdwiltfong** [typehero/typehero#1369]
### Description

At the moment, debugging A11Y violations in TypeHero can be a little cumbersome. For example the axe-playwright plugin displays violations as an array of, rather large, objects. Although the objects provide useful information, such as the HTML that is causing the violation, and a link to dequeuniversity that explains the violation, it also has a lot of noise. On top, because TypeHero uses Next.js, some investigation is needed to connect the violating HTML string to the respective react component. 

It would be great if the axe-plugin would provide the violating component (not just the HTML), but considering playwright analyzes the front-end after it's compiled, it might not be possible. Further investigation is needed. At minimum, TypeHero's integration should implement a UI-mode, and better formatting when detecting A11Y violations. 

```
    1) [chromium] › unauthenticated/home.test.ts:17:3 › homepage a11y › homepage should not have any automatically detectable WCAG A or AA violations in light mode 
web:test:e2e: 
web:test:e2e:     Error: expect(received).toEqual(expected) // deep equality
web:test:e2e: 
web:test:e2e:     - Expected  -  1
web:test:e2e:     + Received  + 56
web:test:e2e: 
web:test:e2e:     - Array []
web:test:e2e:     + Array [
web:test:e2e:     +   Object {
web:test:e2e:     +     "description": "Ensures the contrast between foreground and background colors meets WCAG 2 AA minimum contrast ratio thresholds",
web:test:e2e:     +     "help": "Elements must meet minimum color contrast ratio thresholds",
web:test:e2e:     +     "helpUrl": "https://dequeuniversity.com/rules/axe/4.8/color-contrast?application=playwright",
web:test:e2e:     +     "id": "color-contrast",
web:test:e2e:     +     "impact": "serious",
web:test:e2e:     +     "nodes": Array [
web:test:e2e:     +       Object {
web:test:e2e:     +         "all": Array [],
web:test:e2e:     +         "any": Array [
web:test:e2e:     +           Object {
web:test:e2e:     +             "data": Object {
web:test:e2e:     +               "bgColor": "#efeff0",
web:test:e2e:     +               "contrastRatio": 1.74,
web:test:e2e:     +               "expectedContrastRatio": "4.5:1",
web:test:e2e:     +               "fgColor": "#b7b7b7",
web:test:e2e:     +               "fontSize": "9.0pt (12px)",
web:test:e2e:     +               "fontWeight": "bold",
web:test:e2e:     +               "messageKey": null,
web:test:e2e:     +             },
web:test:e2e:     +             "id": "color-contrast",
web:test:e2e:     +             "impact": "serious",
web:test:e2e:     +             "message": "Element has insufficient color contrast of 1.74 (foreground color: #b7b7b7, background color: #efeff0, font size: 9.0pt (12px), font weight: bold). Expected contrast ratio of 4.5:1",
web:test:e2e:     +             "relatedNodes": Array [
web:test:e2e:     +               Object {
web:test:e2e:     +                 "html": "<div class=\"flex w-fit items-center justify-center rounded-full bg-transparent bg-zinc-300 px-2.5 py-1 pl-1.5 pr-2 text-xs font-bold text-neutral-700 duration-300 group-hover/card:-translate-x-1 group-hover/card:-translate-y-1 group-hover/card:shadow-[0.5rem_0.5rem_0.25rem_-0.25rem_#0004] dark:bg-zinc-700 dark:text-white\">",
web:test:e2e:     +                 "target": Array [
web:test:e2e:     +                   "div:nth-child(2) > .group\\/card.w-\\[320px\\].border-neutral-300 > .gap-3.flex > .bg-zinc-300.pl-1\\.5.dark\\:bg-zinc-700",
web:test:e2e:     +                 ],
web:test:e2e:     +               },
web:test:e2e:     +             ],
web:test:e2e:     +           },
web:test:e2e:     +         ],
web:test:e2e:     +         "failureSummary": "Fix any of the following:
web:test:e2e:     +   Element has insufficient color contrast of 1.74 (foreground color: #b7b7b7, background color: #efeff0, font size: 9.0pt (12px), font weight: bold). Expected contrast ratio of 4.5:1",
web:test:e2e:     +         "html": "<div class=\"flex w-fit items-center justify-center rounded-full bg-transparent bg-zinc-300 px-2.5 py-1 pl-1.5 pr-2 text-xs font-bold text-neutral-700 duration-300 group-hover/card:-translate-x-1 group-hover/card:-translate-y-1 group-hover/card:shadow-[0.5rem_0.5rem_0.25rem_-0.25rem_#0004] dark:bg-zinc-700 dark:text-white\">",
web:test:e2e:     +         "impact": "serious",
web:test:e2e:     +         "none": Array [],
web:test:e2e:     +         "target": Array [
web:test:e2e:     +           "div:nth-child(2) > .group\\/card.w-\\[320px\\].border-neutral-300 > .gap-3.flex > .bg-zinc-300.pl-1\\.5.dark\\:bg-zinc-700",
web:test:e2e:     +         ],
web:test:e2e:     +       },
web:test:e2e:     +     ],
web:test:e2e:     +     "tags": Array [
web:test:e2e:     +       "cat.color",
web:test:e2e:     +       "wcag2aa",
web:test:e2e:     +       "wcag143",
web:test:e2e:     +       "TTv5",
web:test:e2e:     +       "TT13.c",
web:test:e2e:     +       "EN-301-549",
web:test:e2e:     +       "EN-9.1.4.3",
web:test:e2e:     +       "ACT",
web:test:e2e:     +     ],
web:test:e2e:     +   },
web:test:e2e:     + ]
```
